cirq-core/cirq/study/result.py

```diff
@@ -120,6 +119,16 @@ def data(self) -> pd.DataFrame:
         for the measurement key in that repetition. To convert these ints to
         bits see `cirq.big_endian_int_to_bits` and similar functions.
         """
+        # Convert to a DataFrame with columns as measurement keys, rows as
+        # repetitions and a big endian integer for individual measurements.
+        converted_dict = {
```

Comment:
My concern with this is that it's not `@property`-like behavior. `result.data` looks like a simple variable reference, which can conceal performance issues if it's actually doing this every time.

An alternative I could get behind: move this to e.g. `Result._get_data(self)`, have this method remain abstract, and let subtypes call `_get_data` to memoize.

Reply:
Added a staticmethod `datafrom_from_measurements` instead of doing this in a property.